core/src/storage2/lazy/lazy_cell.rs

```diff
@@ -307,6 +308,33 @@ where
     pub fn get_mut(&mut self) -> Option<&mut T> {
         self.load_entry_mut().value_mut().into()
     }
+
+    /// Sets the value in this cell to `value`, without executing any reads.
+    ///
+    /// # Note
+    ///
+    /// No loads will be executed.
+    ///
+    /// # Panics
+    ///
+    /// If accessing the inner value fails.
+    pub fn set(&mut self, new_value: T) {
+        // SAFETY: This is critical because we mutably access the entry.
+        let cache = unsafe { &mut *self.cache.get_ptr().as_ptr() };
+        if cache.is_none() {
+            // Write value to storage
+            let key = self
+                .key
+                .expect("key must exist when attempting to set a value");
+            push_spread_root_opt::<T>(Some(&new_value), &key);
```

Comment:
The `set` operation just as any other non-`PushSpread` operation should never directly write to the contract storage. This is subject to the setup and tear-down process of a contract. So what you want upon `set` if the cache is already populated is to simply overwrite its already existing value. If the cache is empty you simply set the cache to the value. This also means that the `key` does not need to exist for `set` to work since it only needs to write the value into its cache and is done.